src/xmerl_dsig.erl

```diff
@@ -0,0 +1,730 @@
+%% the cert is issued by one of the IssuerCerts in this list.
+%%
+%% Will throw badmatch errors if you give it XML that is not signed
+%% according to the xml-dsig spec. If you're using something other
+%% than rsa+sha1 or sha256 this will asplode. Don't say I didn't warn you.
+%%
+%% This function has now been modified to do Skatteverket's funky removing of
+%% all namespace prefixes before doing a digest. Also there is function
+%% that recreates bugs, which can be removed if you are not implementing
+%% the Skatteverket API.
+%% It will now verify all signatures inside the SOAP structure.
+-spec verify(Element :: #xmlElement{},
+             Fingerprints :: [#'OTPCertificate'{}] | [fingerprint()] | any) ->
+          ok | {error, bad_digest | bad_signature | cert_not_accepted}.
+verify(Element, Fingerprints) ->
+    case xmerl_xpath:string(
+           "//*[\"Signature\"=local-name() and \"http://www.w3.org/2000/"
+           "09/xmldsig#\"=namespace-uri()]", Element) of
+        [] ->
+            {error, no_signature};
+        Signatures when is_list(Signatures) ->
+            verify_signatures(Signatures, Element, Fingerprints)
+    end.
+
+verify_signatures([Signature | Tail], Element, Fingerprints) ->
+    %% DsNs = [{"ds", 'http://www.w3.org/2000/09/xmldsig#'},
+    %%         {"ec", 'http://www.w3.org/2001/10/xml-exc-c14n#'}],
+    [#xmlAttribute{value = SignatureMethodAlgorithm}] =
+        xmerl_xpath:string(
+          "//*[\"SignatureMethod\"=local-name() and \"http://www.w3.org/2000/09/"
+          "xmldsig#\"=namespace-uri()]/@Algorithm", Signature),
+    {HashFunction, _, _} = signature_props(SignatureMethodAlgorithm),
+
+    [#xmlAttribute{value = "http://www.w3.org/2001/10/xml-exc-c14n#"}] =
+        xmerl_xpath:string(
+          "//*[\"CanonicalizationMethod\"=local-name() and \"http://www.w3.org"
+          "/2000/09/xmldsig#\"=namespace-uri()]/@Algorithm", Signature),
+    [#xmlAttribute{value = SignatureMethodAlgorithm}] =
+        xmerl_xpath:string(
+          "//*[\"SignatureMethod\"=local-name() and \"http://www.w3.org/2000/"
+          "09/xmldsig#\"=namespace-uri()]/@Algorithm", Signature),
+    [_C14nTx = #xmlElement{}] =
+        xmerl_xpath:string(
+          "//*[\"Transform\"=local-name() and \"http://www.w3.org/2000/09/"
+          "xmldsig#\"=namespace-uri()]", Signature),
+    TransformAlgo =
+        xmerl_xpath:string(
+          "//*[\"Transform\"=local-name() and \"http://www.w3.org/2000/09/"
+          "xmldsig#\"=namespace-uri()]/@Algorithm", Signature),
+
+    %% We haven't seen InclusiveNs in calls from the Skatteverket API.
+    %% _InclNs = case xmerl_xpath:string("ec:InclusiveNamespaces/@PrefixList",
+    %%                                  C14nTx, [{namespace, DsNs}]) of
+    %%              %% FIXME! -^
+    %%              [] -> [];
+    %%              [#xmlAttribute{value = NsList}] -> string:tokens(NsList, " ,")
+    %%          end,
```

Comment:
What does this mean for the library user? Will some verification crash, or not happen instead?